spec/tunes/app_version_spec.rb

```diff
@@ -150,10 +171,308 @@
+
+        expect(trailer.video_asset_token).to eq(nil)
+        expect(trailer.picture_asset_token).to eq("Purple70/v4/5f/2b/81/5f2b814d-1083-5509-61fb-c0845f7a9374/pr_source.jpg")
+        expect(trailer.descriptionXML).to eq(nil)
+        expect(trailer.preview_frame_time_code).to eq("00:00:06:12")
+        expect(trailer.video_url).to eq("http://a1713.phobos.apple.com/us/r30/PurpleVideo7/v4/be/38/db/be38db8d-868a-d442-87dc-cb6d630f921e/P37134684_default.m3u8")
+        expect(trailer.preview_image_url).to eq("https://is1-ssl.mzstatic.com/image/thumb/PurpleVideo5/v4/b7/41/5e/b7415e96-5ad5-6cf5-9323-15122145e53f/Job21976428-61a9-456b-af46-26c1303ae607-91524171-PreviewImage_AppTrailer_quicktime-Time1438426738374.png/500x500bb-80.png")
+        expect(trailer.full_sized_preview_image_url).to eq("https://is1-ssl.mzstatic.com/image/thumb/PurpleVideo5/v4/b7/41/5e/b7415e96-5ad5-6cf5-9323-15122145e53f/Job21976428-61a9-456b-af46-26c1303ae607-91524171-PreviewImage_AppTrailer_quicktime-Time1438426738374.png/900x1200ss-80.png")
+        expect(trailer.device_type).to eq("ipad")
+        expect(trailer.language).to eq("English")
+      end
+
+      it "can add a new trailer given a valid externally provided preview screenshot" do
+        # remove existing
+        version.upload_trailer!(nil, "English", 'ipad')
+
+        expect do
+          version.upload_trailer!(ipad_trailer_path, "English", 'ipad', '12.34', ipad_external_invalid_trailer_preview_path)
+        end.to raise_error "Invalid portrait screenshot resolution for device ipad. Should be [768, 1024]"
+      end
+
+      it "can add a new trailer given a valid externally provided preview screenshot" do
+        # remove existing
+        version.upload_trailer!(nil, "English", 'ipad')
+
+        count = ipad_trailers.count
+        expect(count).to eq(0)
+        version.upload_trailer!(ipad_trailer_path, "English", 'ipad', '12.34', ipad_external_valid_trailer_preview_path)
+        count_after = ipad_trailers.count
+        expect(count_after).to eq(count + 1)
+        trailer = ipad_trailers[0]
+        expect(trailer.video_asset_token).to eq("VideoSource40/v4/e3/48/1a/e3481a8f-ec25-e19f-5048-270d7acaf89a/pr_source.mov")
+        expect(trailer.picture_asset_token).to eq("Purple69/v4/5f/2b/81/5f2b814d-1083-5509-61fb-c0845f7a9374/pr_source.jpg")
+        expect(trailer.descriptionXML).to match(/FoghornLeghorn/)
+        expect(trailer.preview_frame_time_code).to eq("00:00:12:34")
+        expect(trailer.video_url).to eq(nil)
+        expect(trailer.preview_image_url).to eq(nil)
+        expect(trailer.full_sized_preview_image_url).to eq(nil)
+        expect(trailer.device_type).to eq("ipad")
+        expect(trailer.language).to eq("English")
+      end
+
+      # TODO: can we detect trailer source change ?
+
+      it "remove the video trailer" do
+        count = ipad_trailers.count
+        expect(count).to eq(1)
+        version.upload_trailer!(nil, "English", 'ipad')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.